common/src/main/java/com/revenuecat/purchases/common/OfferingFactories.kt

```diff
@@ -50,16 +50,19 @@ fun JSONObject.createOffering(productsById: Map<String, StoreProduct>): Offering
 }
 
 fun JSONObject.createPackage(
-    productsById: Map<String, StoreProduct>,
+    productsById: Map<String, List<StoreProduct>>,
     offeringIdentifier: String
 ): Package? {
     val packageIdentifier = getString("identifier")
     val productGroupIdentifier = optString("platform_product_group_identifier")
+    // TODO do we care about productIdentifier? that was used for BC4, not sure it matters now... 
     val productIdentifier = getString("platform_product_identifier")
     val productDuration = optString("product_duration")
-    return productsById[productGroupIdentifier]?.let { product ->
+    val subProducts = productsById[productGroupIdentifier]
+    val matchingProduct = subProducts?.firstOrNull { it.subscriptionPeriod == productDuration }
```

Comment:
I think this is broken for consumables right?

Reply:
yeah :/  i wonder if we should make a separate ticket for consumables support?

Reply:
maybe if the backend only has productIdentifier we assume it's a consumable?

Reply:
yes... I am not sure what the backend sends down for duration? But we should use that to determine if it's a consumable or not

Reply:
@swehner fyi, idk if we already had a ticket for consumables? might involve some backend too... i'll add it to the open questions section